geopandas_view/view.py

```diff
@@ -478,24 +496,142 @@ def _choropleth(
         bins = binning.bins.tolist()
         bins.insert(0, gdf[column].min())
 
-    choro = folium.Choropleth(
-        gdf.__geo_interface__,
-        data=gdf[["__folium_key", column]],
-        key_on="feature.properties.__folium_key",
-        columns=["__folium_key", column],
-        legend_name=column,
-        fill_color=cmap,
-        bins=bins,
-        name=column,
-        **kwds,
-    )
+    geo_data = gdf.__geo_interface__
+    data = gdf[columns]
+
+    # Create color_data dict
+    if hasattr(data, "set_index"):
+        # This is a pd.DataFrame
+        color_data = data.set_index("__folium_key")[column].to_dict()
+    elif hasattr(data, "to_dict"):
+        # This is a pd.Series
+        color_data = data.to_dict()
+    elif data:
+        color_data = dict(data)
+    else:
+        color_data = None
```

Comment:
```suggestion
    else:
        # This is a pd.Series
        color_data = data.to_dict()
```

I may be missing something obvious here, but `data` is defined as `gdf[columns]` so it is always either DataFrame of Series, isn't it? Why do you have these other cases here?